extra/register-service.rb

```diff
@@ -1,52 +1,49 @@
-#!/usr/bin/env ruby
-
 if RUBY_PLATFORM !~ /mingw/
   puts "To install this service on Unix please create a startup script"
-else
-  require 'rubygems'
-  require 'highline/import'
-  require 'win32/service'
-  require 'rbconfig'
-  include Win32
-  include Config
+  exit
+end
 
-  executable = Pathname.new(__FILE__).dirname.parent.join("bin", "smart-proxy-win-service")
-  executable = executable.realpath
-  ruby = File.join(CONFIG['bindir'], 'ruby').tr('/', '\\')
-  cmd  = ruby + ' "' + executable.to_s.tr('/', '\\') + '"'
-  puts "Installing #{cmd} as a service"
+require 'rubygems'
+require 'highline/import'
+require 'win32/service'
+require 'rbconfig' if RUBY_VERSION < '1.9'
 
-  default_user = ENV["USERNAME"]
-  default_user = ENV["USERDOMAIN"] + '\\' + default_user if ENV["USERDOMAIN"]
+include Win32
+include RbConfig
 
-  puts "This service must run as a user with permission to execute the netsh dhcp script"
-  puts 'The acount can be local or a domain account. If it is a domain account then use the domain\account syntax'
-  user  = ask("Run this service as this user? ") {|u| u.default = default_user}
-  pass1 = ask("Enter the user's password ") {|p| p.echo = "x"}
-  pass2 = ask("Reenter the password "){|p| p.echo = "x"}
-  unless pass1 == pass2
-    print "Passwords did not match!"
-    exit 1
-  end
+executable = Pathname.new(__FILE__).dirname.parent.join("bin", "smart-proxy-win-service")
+executable = executable.realpath.to_s.tr('/', '\\')
+working_dir = Pathname.new(__FILE__).dirname.parent.realpath
+ruby = File.join(CONFIG['bindir'], 'ruby').tr('/', '\\')
+cmd  = "#{ruby} -C #{working_dir} \"#{executable}\""
+puts "Installing #{cmd} as a service"
+
+default_user = ENV["USERNAME"]
+default_user = ENV["USERDOMAIN"] + '\\' + default_user if ENV["USERDOMAIN"]
+
+puts "This service must run as a user with permission to execute the netsh dhcp script"
+puts 'The acount can be local or a domain account. If it is a domain account then use the domain\account syntax'
+user  = ask("Run this service as this user? ") {|u| u.default = default_user}
+pass1 = ask("Enter the user's password ") {|p| p.echo = "x"}
+
+begin
+  Service.stop("smart proxy")   rescue nil
```

Comment:
We should safe ourselves some hassle and use `smart_proxy` or `smart-proxy` here